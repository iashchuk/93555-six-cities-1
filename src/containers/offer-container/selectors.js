import { createSelector } from "reselect";

export const getOfferNeighbourhoods = createSelector(
    (f) => f,
    ([offers, offer]) => {
      if (!offer) {
        return [];
      }
      return offers.get(offer.city).slice(0, 3) || [];
    }
);
