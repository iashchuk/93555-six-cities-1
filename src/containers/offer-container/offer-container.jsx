import React from "react";
import { connect } from "react-redux";

import Offer from "../../components/offer/offer.jsx";
import { getOfferAsync, getCommentsAsync } from "../../redux/offer/actions";
import { setFavoriteAsync } from "../../redux/hotels/actions";
import { sendReviewAsync } from "../../redux/offer/actions";
import { getOfferNeighbourhoods } from "./selectors.js";

export const OfferContainer = (props) => {
  return <Offer {...props} />;
};

const mapStateToProps = ({ offer, hotels, auth }) => {
  return {
    isAuthenticated: auth.isAuthenticated,
    isFormSending: offer.isFormSending,
    offer: offer.currentOffer,
    offers: getOfferNeighbourhoods([hotels.offers, offer.currentOffer]),
    comments: offer.comments,
    sendFormError: offer.errors.sendForm
  };
};

const mapDispatchToProps = {
  getOfferAsync,
  getCommentsAsync,
  setFavoriteAsync,
  sendReviewAsync
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(OfferContainer);
