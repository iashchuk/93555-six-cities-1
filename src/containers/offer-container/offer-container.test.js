import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { OfferContainer } from "./offer-container.jsx";

configure({ adapter: new Adapter() });

const isAuthenticated = true;
const getOffer = jest.fn();
const getComments = jest.fn();
const setFavoriteAsync = jest.fn();
const sendReviewAsync = jest.fn();

describe(`OfferContainer`, () => {
  it(`renders correctly`, () => {
    const component = shallow(
        <OfferContainer
          isAuthenticated={isAuthenticated}
          getOffer={getOffer}
          getComments={getComments}
          setFavoriteAsync={setFavoriteAsync}
          sendReviewAsync={sendReviewAsync}
        />
    );
    expect(component).toMatchSnapshot();
  });
});
