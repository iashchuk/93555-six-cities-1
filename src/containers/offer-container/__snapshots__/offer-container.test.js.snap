// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferContainer renders correctly 1`] = `ShallowWrapper {}`;
